.github/workflows/e2e-nvidia-t4-x1.yml

```diff
@@ -121,3 +126,12 @@ jobs:
           github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
           label: ${{ needs.start-runner.outputs.label }}
           ec2-instance-id: ${{ needs.start-runner.outputs.ec2-instance-id }}
+
+  e2e-workflow-complete:
+    # we don't want to block PRs on failed EC2 cleanup
+    # so not requiring "stop-runner" as well
+    needs: ["start-runner", "e2e"]
```

Comment:
You only need `e2e` here since `e2e` itself depends upon `start-runner`. And in that case, there is no need then for this workflow complete job since `e2e` is sufficient.